src/Application/FileProcessor.php

```diff
@@ -57,6 +59,8 @@ public function processFile(File $file, Configuration $configuration): FileProce
             return new FileProcessResult([$parsingSystemError], null);
         }
 
+        $this->fileStorage->addFile($file->getFilePath(), $file);
```

Comment:
`clearFile()` after `do {}` while may be needed, otherwise, more and more data collected into storage when go to another file  that no longer needed, saved into memory...

I prefer to not have indexed by file path data, instead, just nullable File object so it can easily be cleared after all processed.

Reply:
I tested locally, memory increased overtime, from 4GB to 6GB for rector-src codebase.